packages/astro/src/core/app/index.ts

```diff
@@ -232,7 +282,13 @@ export class App {
 			status,
 		});
 
-		const result = await callEndpoint(handler, this.#env, ctx, this.#logging);
+		const result = await callEndpoint(
+			handler,
+			this.#env,
+			ctx,
+			this.#logging,
+			this.#manifest.middleware as AstroMiddlewareInstance<Response | EndpointOutput>
```

Comment:
Isn't middleware going to be `undefined` if the feature is not enabled?

Reply:
I solved the issue without casting